go-runtime/server/server.go

```diff
@@ -67,10 +67,14 @@ func Handle[Req, Resp any](verb func(ctx context.Context, req Req) (Resp, error)
 				return nil, fmt.Errorf("call to verb %s failed: %w", ref, err)
 			}
 
+			fmt.Printf("resp: %v\n", resp)
+
 			respdata, err := encoding.Marshal(resp)
 			if err != nil {
 				return nil, err
 			}
+
+			fmt.Printf("respdata: %v\n", string(respdata))
```

Comment:
@alecthomas I might need to pick your brain on this one. It seems like when we encode arrays of strings, they are getting double quoted here. Not sure if that's what we want, or if I need to make a change to the encoder. It also causes code after this to fail since it can't probably json encode the response for `recordCall` in `observability`.

This is what is printed out when I run the integration tests for HTTP Ingress.
```bash
resp: {0 map[] ["hello" "world"] None}
respdata: {"status":0,"headers":{},"body":[""hello"",""world""],"error":null}
```

Reply:
Use `repr.Print()` as it will tell you what the values actually are. `%v` is useless.

Reply:
ah yeah!

resp:
```
builtin.HttpRequest[[]string]{
  Method: "GET",
  Path: "/array/string",
  PathParameters: map[string]string{  },
  Query: map[string][]string{  },
  Headers: map[string][]string{
    "Accept-Encoding": []string{
      "gzip",
    },
    "Content-Length": []string{
      "17",
    },
    "Content-Type": []string{
      "application/json",
    },
    "User-Agent": []string{
      "Go-http-client/1.1",
    },
  },
  Body: []string{
    "\"hello\"",
    "\"world\"",
  },
}
```

respdata:
```
[]byte("{\"status\":0,\"headers\":{},\"body\":[\"\"hello\"\",\"\"world\"\"],\"error\":null}")
```

Reply:
Is that before or after? Whatever it is, it should tell you that either the strings are coming in already with extra quotes, or they're getting added by encode. That should point you closer to where the problem lies.

Reply:
Thanks for the heads up on this one! The issue was that the data in the `req` was already encoded incorrectly and since it's just returned in the `Body: req.Body` the response was also encoded incorrectly. Garbage in, garbage out. 🤦